cmd/link.go

```diff
@@ -38,28 +39,38 @@ printed to STDOUT.
 		checkError(err, "")
 		projectName := project.GetCurrent(client)
 
-		sourceComponent := getComponent(client, linkComponent, applicationName, projectName)
-		targetComponent := args[0]
+		componentName := getComponent(client, linkComponent, applicationName, projectName)
+		serviceName := args[0]
 
-		exists, err := component.Exists(client, sourceComponent, applicationName, projectName)
+		exists, err := component.Exists(client, componentName, applicationName, projectName)
 		checkError(err, "")
 		if !exists {
-			fmt.Printf("Component %v does not exist\n", sourceComponent)
+			fmt.Printf("Component %v does not exist\n", componentName)
+			os.Exit(1)
 		}
-		exists, err = component.Exists(client, targetComponent, applicationName, projectName)
-		checkError(err, "")
+
+		exists, err = svc.SvcExists(client, serviceName, applicationName, projectName)
+		checkError(err, "Unable to determine if Service %s exists within the current namespace", serviceName)
 		if !exists {
-			fmt.Printf("Component %v does not exist\n", targetComponent)
+			fmt.Printf(`Service %s doesn't exist within the current namespace.
+Please perform 'odo service create %s ...' before attempting to link the service.`, serviceName, serviceName)
+			os.Exit(1)
 		}
 
-		linkInfo, err := component.Link(client, sourceComponent, targetComponent, applicationName)
-		checkError(err, fmt.Sprintf("Failed to link %v to %v", targetComponent, sourceComponent))
-
-		fmt.Printf("Successfully linked %v to %v\n", targetComponent, sourceComponent)
-		fmt.Printf("The following environment variables have been injected in %v to connect to %v\n", linkInfo.SourceComponent, linkInfo.TargetComponent)
-		for _, env := range linkInfo.Envs {
-			fmt.Printf("- %v\n", env)
+		// we also need to check whether there is a secret with the same name as the service
+		// the secret should have been created along with the secret
+		_, err = client.GetSecret(serviceName, projectName)
+		if err != nil {
+			fmt.Printf(`Secret %s should have been created along with the service!
+If you previously created the service with 'odo service create...', then you might have to wait a few seconds until Openshift provisions it.
```

Comment:
`If you previously created the service with 'odo service create', then you may have to wait a few seconds until OpenShift provisions it.`

Remove `...`, need to capitalize `OpenShift`